src/main/java/world/bentobox/bentobox/listeners/PortalTeleportationListener.java

```diff
@@ -92,202 +105,256 @@ public void onPlayerPortal(PlayerMoveEvent e) {
+     * @param env - environment
+     * @return true or false
+     */
+    private boolean isMakePortals(GameModeAddon gm, Environment env) {
+        return env.equals(Environment.NETHER) ? gm.getWorldSettings().isMakeNetherPortals() : gm.getWorldSettings().isMakeEndPortals();
+    }
+
+    /**
+     * Check if nether or end are generated
+     * @param gm - game mode
+     * @param env - environment
+     * @return true or false
+     */
+    private boolean isGenerate(World overWorld, Environment env) {
+        return env.equals(Environment.NETHER) ? plugin.getIWM().isNetherGenerate(overWorld) : plugin.getIWM().isEndGenerate(overWorld);
+    }
+
+    /**
+     * Check if nether or end islands are generated
+     * @param gm - game mode
+     * @param env - environment
+     * @return true or false
+     */
+    private boolean isIslands(World overWorld, Environment env) {
+        return env.equals(Environment.NETHER) ? plugin.getIWM().isNetherIslands(overWorld) : plugin.getIWM().isEndIslands(overWorld);
+    }
+
+    /**
+     * Get the nether or end world
+     * @param gm - game mode
+     * @param env - environment
+     * @return nether or end world
+     */
+    private World getNetherEndWorld(World overWorld, Environment env) {
+        return env.equals(Environment.NETHER) ? plugin.getIWM().getNetherWorld(overWorld) : plugin.getIWM().getEndWorld(overWorld);
+    }
+
+    /**
+     * Check if the island has a nether or end island already
+     * @param gm - game mode
+     * @param env - environment
+     * @return true or false
+     */
+    private boolean hasPartnerIsland(Island i, Environment env) {
+        return env.equals(Environment.NETHER) ? i.hasNetherIsland() : i.hasEndIsland();
+    }
+
+    /**
+     * Check if the default nether or end are allowed by the server settings
+     * @param gm - game mode
+     * @param env - environment
+     * @return true or false
+     */
+    private boolean isAllowedOnServer(Environment env) {
+        return env.equals(Environment.NETHER) ? Bukkit.getAllowNether() : Bukkit.getAllowEnd();
+    }
+
+    /**
+     * Handle teleport from nether or end to overworld
+     * @param e - event
+     * @param overWorld - over world
+     * @param env - environment
+     */
+    private void handleFromNetherOrEnd(PlayerEntityPortalEvent e, World overWorld, Environment env) {
+        // Standard portals
+        if (plugin.getIWM().getAddon(overWorld).map(gm -> isMakePortals(gm, env)).orElse(false)) {
+            e.setTo(e.getFrom().toVector().toLocation(overWorld));
+            // Find distance from edge of island's protection
+            plugin.getIslands().getIslandAt(e.getFrom()).ifPresent(i -> setSeachRadius(e, i));
+            inPortal.remove(e.getEntity().getUniqueId());
+            return;
+        }
+        // Custom portals
+        e.setCancelled(true);
+        // If this is from the island nether or end, then go to the same vector, otherwise try island home location
```

Comment:
The comment for the code below is telling different than the actual code. One of them is wrong.